src/cycax/cycad/assembly.py

```diff
@@ -186,9 +213,10 @@ def _final_place_(self, part):
         """
         for hole in range(len(part.move_holes)):
```

Comment:
FYI: Python has a builtin for this: Please see https://docs.python.org/3/library/functions.html#enumerate

```
for hole_i, hole in enumerate(part.move_holes):
  # Then hole_i is the list id and hole is the value at that hole_i
  temp_hole = copy.deepcopy(hole)
  temp_hole =  # The code thats there 
  part.move_holes[hole_i] = temp_hole # Line 226
```
 
Note: in my example hole is the data and hole_i is the counter, in your code hole is the counter.